rust/processor/src/processors/token_v2_processor.rs

```diff
@@ -605,18 +605,21 @@ async fn parse_v2_token(
                 if let Some(burn_event) = BurnEvent::from_event(event, txn_version).unwrap() {
                     tokens_burned.insert(burn_event.get_token_address());
                 }
-                if let Some(transfer_event) = TransferEvent::from_event(event, txn_version).unwrap()
+                if let Some(transfer_events) =
+                    TransferEvent::from_event(event, txn_version).unwrap()
                 {
                     if let Some(aggregated_data) =
-                        token_v2_metadata_helper.get_mut(&transfer_event.get_object_address())
+                        token_v2_metadata_helper.get_mut(&transfer_events.get_object_address())
                     {
                         // we don't want index to be 0 otherwise we might have collision with write set change index
                         let index = if index == 0 {
-                            user_txn.events.len()
+                            user_txn.events.len() - aggregated_data.transfer_events.len()
```

Comment:
won't this potentially collide with wsc index?

Reply:
No b/c we set those to negative. 0 obviously will still be 0 hence the conflict.